### PR TITLE
Issue 10

### PR DESCRIPTION
**Pull request by rameezjhaveri**
- [x] Merge is going to correct branch.
- [x] Code is well formated (readable)
- [x] Code uses sensible variable/method names (readable)
- [x] Peer Reviewed
- [x] Reviewed by Tadas

### CODE REVIEW

File: app/controllers/LogoutController.scala [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+
+import scala.concurrent.Future
+
+class LogoutController @Inject()(val messagesApi: MessagesApi, val materializer: Materializer) extends Controller
+  with I18nSupport {
+
+  def logout: Action[AnyContent] = Action.async { implicit request =>
+    Future{
+      Redirect(routes.Application.index()).withSession()
```
Comment (phtb233) [inline app/controllers/LogoutController.scala]: could use .withNewSession instead of .withSession

File: app/views/login.scala.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+@main("Login") {
+
+    <div>
+        <h2>Please log in</h2>
+    </div>
+
+    <div>
+    @helper.form(routes.LoginController.loginSubmit()){
+        @flash.get("failed").getOrElse("")
+        @helper.inputText(form("username"), '_label -> "Login", '_showConstraints -> false)
```
Comment (phtb233) [inline app/views/login.scala.html]: 'Username' would make the label clearer instead of 'Login'

File: app/controllers/LoginController.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    Future{
+      Ok(views.html.login(LoginDetails.loginForm))
+    }
+  }
+
+  def loginSubmit: Action[AnyContent] = Action.async { implicit request =>
+    var loginDetails = LoginDetails(Constants.emptyString.toString, Constants.emptyString.toString)
+
+    LoginDetails.loginForm.bindFromRequest.fold({
+      formWithErrors => {
```
Comment (phtb233) [inline app/controllers/LoginController.scala]: 'formWithErrors' could be a '_' as its not used

File: app/models/LoginDetails.scala [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      Valid
+    }
+    else {
+      Invalid(ValidationError(""))
+    }
+  })
+
+  def validUser(loginDetails: LoginDetails): Boolean = {
+    signUps.count(result =>
+      if (result.username.equals(loginDetails.username) && result.password.equals(loginDetails.password)) {
```
Comment (phtb233) [inline app/models/LoginDetails.scala]: the predicate for 'count' could be simplified:
`signUps.exists(result => result.username.equals(loginDetails.username) && result.password.equals(loginDetails.password))`

File: app/models/SignUp.scala [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        .verifying("too many chars", s => lengthIsLessThanNCharacters(s, 30))
+    )(SignUp.apply)(SignUp.unapply)
+  )
+  val signUps: ArrayBuffer[SignUp] = ArrayBuffer(
+    SignUp("Fabian", "Lewis", "Fab", "password"),
+    SignUp("Rameez", "J", "Rico", "password"),
+    SignUp("Steven", "B", "Steve", "password")
+  )
+  private def lengthIsGreaterThanNCharacters(s: String, n: Int): Boolean = {
+    if (s.length > n) true else false
```
Comment (phtb233) [inline app/models/SignUp.scala]: 'lengthIsGreaterThanNCharacters' and 'lengthIsLessThanNCharacters' could be simplified: 
`private def lengthIsGreaterThanNCharacters(s: String, n: Int): Boolean = s.length > n`
`private def lengthIsLessThanNCharacters(s: String, n: Int): Boolean = s.length < n`

**Review by phtb233 — CHANGES_REQUESTED** [review]

**Review by phtb233 — APPROVED** [review]

File: app/controllers/Application.scala [hunk 6]
```diff
@@ -1,14 +1,17 @@
...
 
 class Application extends Controller {
 
-  def index: Action[AnyContent] = Action {
-    request =>
-      request.flash.get(Constants.login.toString).map(welcomeMessage => Ok(views.html.index(welcomeMessage)))
-        .getOrElse(Ok(views.html.index(Constants.indexString.toString)))
+  def index: Action[AnyContent] = Action { request =>
+    request.session.get(Constants.username.toString).map{ user =>
+      Ok(views.html.index(Constants.indexString.toString +" "+ user))
```
Comment (tvaidotas) [inline app/controllers/Application.scala]: `" "` put it into constants file

File: app/controllers/LoginController.scala [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      Ok(views.html.login(LoginDetails.loginForm))
+    }
+  }
+
+  def loginSubmit: Action[AnyContent] = Action.async { implicit request =>
+    var loginDetails = LoginDetails(Constants.emptyString.toString, Constants.emptyString.toString)
+
+    LoginDetails.loginForm.bindFromRequest.fold({
+      _ => {
+        Future(Redirect(routes.LoginController.login()).flashing("failed" -> "Incorrect username or password"))
```
Comment (tvaidotas) [inline app/controllers/LoginController.scala]: `"failed"` and `"Incorrect username or password"`

File: app/controllers/UserController.scala [hunk 8]
```diff
@@ -1,19 +1,21 @@
...
+  def account: Action[AnyContent] = AuthenticatedAction { request =>
+    Ok(views.html.account(request.session.get(Constants.username.toString).getOrElse(Constants.emptyString.toString)))
 
 
+  }
 
-}
\ No newline at end of file
+}
+//s"Hello ${request.session.get("username").getOrElse(Constants.emptyString.toString)}")
```
Comment (tvaidotas) [inline app/controllers/UserController.scala]: why is this line here?
